config/app.default.php

```diff
@@ -232,7 +237,14 @@
             'username' => 'my_app',
             'password' => 'secret',
             'database' => 'my_app',
-            'encoding' => 'utf8',
+            /**
+             * You do not need to set this flag to use full utf-8 encoding (internal default since CakePHP 3.6).
+             * If you do these are the recommendations:
+             * 'encoding' => 'utf8mb4' // On MariaDB/MySQL for full UTF-8 encoding (4-byte)
+             * 'encoding' => 'utf8mb3' // On MariaDB/MySQL for limited UTF-8 encoding (3-byte, not recommended)
+             * 'encoding' => 'utf8' // On any other RDBMS for full 4-byte UTF-8 encoding (4-byte)
+             */
+            // 'encoding' => 'utf8mb4',
```

Comment:
This line is not necessary with a an inline comment above.

Reply:
Do you agree?

Reply:
PS: note the missing whitespace after // if it is commented out code vs. comment.

Reply:
I thought you would keep the current

     'encoding' => 'utf8',

If you indeed remove the line, then you can of course keep the 2nd commented out one :)
But maybe we should keep the current default as it is valid for 3/4 of the systems?

Reply:
I don't think so.
- Most starters will use mysql/mariadb
- Those are the most likely to just out-comment the block and have broken defaults by default (broken: 3-byte utf8 but you would not know until you encounter run-time problems)
- For the other (more power-)users: They would either read the docblock and not out-comment it, or outcomment and adapt the block. If not they will face an error right in their face like 'utf8mb4 not supported encoding' on postgres and alike and not just encounter it at some point in future without knowing about it.